src/serializer/ResidualReactElementSerializer.js

```diff
@@ -252,7 +273,15 @@ export class ResidualReactElementSerializer {
     if (reactLibraryObject !== undefined) {
       this.residualHeapSerializer.serializeValue(reactLibraryObject);
     }
-    let identifier = convertExpressionToJSXIdentifier(this.residualHeapSerializer.serializeValue(typeValue), true);
+    let identifier;
+    if (typeValue instanceof SymbolValue && typeValue === getReactSymbol("react.fragment", this.realm)) {
+      identifier = convertExpressionToJSXIdentifier(
+        this._serializeReactFragmentType(typeValue, reactLibraryObject),
+        true
+      );
+    } else {
+      identifier = convertExpressionToJSXIdentifier(this.residualHeapSerializer.serializeValue(typeValue), true);
```

Comment:
Nit: this code would be clearer if we kept the function call unconditional, and only branched for the first argument